src/many-server/src/server.rs

```diff
@@ -408,4 +432,46 @@ mod tests {
         // NOT okay with the future too much
         assert!(_validate_time(&request, timestamp + Duration::from_secs(101), 100).is_err());
     }
+
+    #[test]
+    fn server_manages_time() {
+        fn create_request(timestamp: SystemTime, nonce: u8) -> CoseSign1 {
+            let request: RequestMessage = RequestMessageBuilder::default()
+                .method("status".to_string())
+                .timestamp(timestamp)
+                .nonce(nonce.to_le_bytes().to_vec())
+                .build()
+                .unwrap();
+            encode_cose_sign1_from_request(request, &CoseKeyIdentity::anonymous()).unwrap()
+        }
+
+        let server = ManyServer::new("test-server", CoseKeyIdentity::anonymous(), None);
+        let timestamp = SystemTime::now();
+        let now = Arc::new(RwLock::new(timestamp.clone()));
+        let get_now = {
+            let n = now.clone();
+            move || Ok(n.read().unwrap().clone())
+        };
+
+        // timestamp is now, so this should be fairly close to it and should pass.
+        assert!(smol::block_on(server.execute(create_request(timestamp, 0))).is_ok());
+
+        // Set time to present.
+        {
+            server.lock().unwrap().set_time_fn(get_now);
+        }
+        assert!(smol::block_on(server.execute(create_request(timestamp, 1))).is_ok());
+
+        // Set time to 10 minutes past.
```

Comment:
Can you also add a test where both `get_time()` AND the request are sometime in the past? This is the case we really want to test with block replay.